### PR TITLE
fix(diagnostics): read speed test history from the archive report

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** `get_speed_test_history` queries a `rest/speedtest` resource
that does not exist; every call fails with `api.err.InvalidObject`.

**Root cause + fix:** results are stored in the
`stat/report/archive.speedtest` report, verified live against Network
10.5.67: entries carry `xput_download`/`xput_upload` in Mbps, `latency`
in ms and `time` in epoch milliseconds. The tool now POSTs the report
query with a caller-selectable window (`hours`, default one week) and
translates entries into its existing result shape, timestamps as
ISO 8601.

**Live verification (Network 10.5.67):** returns the controller's real
stored history for the requested window.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

The new `hours` parameter is optional; the old route never returned
anything.

## Testing

- Fixtures mirror the real archive entries; tests cover the report query
  body (attrs/start/end), unit translation, ISO timestamps, and the
  window parameter.
- `pytest tests/unit/` passes; pre-commit hooks pass for all files this
  PR touches.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; report route verified against a
live Network 10.5.67 controller.
**Security Review:** Completed — no identifying values; fixtures are
synthetic.

### CODE REVIEW

File: src/tools/diagnostics.py [hunk 1]
```diff
@@ -108,28 +110,66 @@ async def get_speed_test_status(site_id: str, settings: Settings) -> dict[str, A
...
             data = []
 
-        results = [SpeedTestResult(**item).model_dump() for item in data]
+        # Archive entries report xput_* in Mbps, latency in ms and time in
+        # epoch milliseconds; translate to this module's result shape.
+        results = []
+        for item in data:
+            if not isinstance(item, dict):
+                continue
+            ts = item.get("time")
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/diagnostics.py:169]: Fixed in 996cbe7 — entries are sorted by their epoch time before translation, so the docstring's oldest-first promise is real.

File: src/tools/diagnostics.py [hunk 2]
```diff
@@ -108,28 +110,66 @@ async def get_speed_test_status(site_id: str, settings: Settings) -> dict[str, A
...
     Returns:
-        List of speed test result dictionaries
+        List of speed test result dictionaries, oldest first
     """
     site_id = validate_site_id(site_id)
     logger = get_logger(__name__, settings.log_level)
 
+    end_ms = int(time.time() * 1000)
+    start_ms = end_ms - hours * 3600 * 1000
+
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/diagnostics.py:143]: Fixed in 996cbe7: `hours` is coerced and validated up front; zero, negative, and non-numeric values raise ValidationError.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
